Source/WebCore/html/HTMLElement.cpp

```diff
@@ -846,7 +843,7 @@ void HTMLElement::dirAttributeChanged(const AtomString& value)
 {
     RefPtr<Element> parent = parentElement();
 
-    if (is<HTMLElement>(parent) && parent->selfOrPrecedingNodesAffectDirAuto())
+    if (is<HTMLElement>(parent) && parent->selfOrPrecedingNodesAffectDirAuto() && isValidDirValue(value))
```

Comment:
Is this new clause really needed for correctness? If not I am not sure we need to optimize this case.

Reply:
Yes, this is required. Otherwise adjustDirectionalityIfNeededAfterChildAttributeChanged would erroneously clear selfOrPrecedingNodesAffectDirAuto flag when an invalid dir attribute is specified.